centreon/www/widgets/src/centreon-widget-resourcestable/src/Listing/Actions/DisplayType/Icons.tsx

```diff
@@ -0,0 +1,51 @@
+import parse from 'html-react-parser';
+import { equals } from 'ramda';
+
+import { SvgIcon, useTheme } from '@mui/material';
+
+import { DisplayType } from '../../models';
+
+interface Props {
+  displayType: DisplayType;
+  isActive: boolean;
+}
+
+const getViewByService = (color, backgroundColor): string =>
+  `<svg width="24" height="24" viewBox="0 0 24 24"><rect width="24" height="24" rx="4" fill="${backgroundColor}"/><g transform="translate(1 2)"><circle id="Ellipse_192" cx="2" cy="2" r="2" transform="translate(2 2)" fill="${color}"/><path id="Tracé_2976" d="M-2.979,0h16" transform="translate(6.979 4)" fill="none" stroke="${color}" stroke-width="2"/></g><g transform="translate(1 2)"><circle id="Ellipse_193" cx="2" cy="2" r="2" transform="translate(2 8)" fill="${color}"/><path id="Tracé_2977" d="M-2.979,0h16" transform="translate(6.979 10)" fill="none" stroke="${color}" stroke-width="2"/></g><g transform="translate(1 3)"><circle id="Ellipse_194" cx="2" cy="2" r="2" transform="translate(2 13)" fill="${color}"/><path id="Tracé_2978" d="M-2.979,0h16" transform="translate(6.979 15)" fill="none" stroke="${color}" stroke-width="2"/></g></svg>`;
+
+const getViewByHost = (color, backgroundColor): string =>
+  `<svg width="24" height="24" viewBox="0 0 24 24"><rect width="24" height="24" rx="4" fill="${backgroundColor}"/><circle cx="3.5" cy="3.5" r="3.5" transform="translate(2 3)" fill="${color}"/><line x2="10" transform="translate(12 20)" fill="none" stroke="${color}" stroke-width="2"/><line x2="10" transform="translate(12 16)" fill="none" stroke="${color}" stroke-width="2"/><line x2="10" transform="translate(12 12)" fill="none" stroke="${color}" stroke-width="2"/><path d="M8.829,6.5h3.317V21" fill="none" stroke="${color}" stroke-width="2"/></svg>`;
+
+const getViewByAll = (color, backgroundColor): string =>
```

Comment:
in place of adding our own props make i separate custom icon using ScgIcon that use the same props as mui

https://mui.com/material-ui/icons/#color

Reply:
We already use SvgIcon from @centreon/ui. These functions are just to change the colors depending on whether the button is activated or not.